src/BaseElement.js

```diff
@@ -294,6 +305,58 @@ class BaseElement extends HTMLElement {
 		}
 	}
 
+	/**
+	 * Context Properties to issue Context Requests and to pull contextual properties into the elements scope
+	 * @return {{}}
+	 */
+	injectProperties() {
+		return {};
+	}
+
+	/**
+	 * Define which properties that can be provided to children
+	 * @return {{}}
+	 */
+	provideProperties() {
+		return {};
+	}
+
+	/**
+	 * Defines context on the element based on keys from this.context()
+	 */
+	definePropertyInjection() {
+		Object.entries(this.injectProperties()).forEach(([key, value]) => {
+			this.requestContext(key, value);
+		});
+		// define context provider
+		if (Object.keys(this.provideProperties()).length > 0) {
+			this.addEventListener('request-context', this.onRequestContext);
+		}
+	}
+
+	requestContext(propertyName, valueOrCallback) {
+		this.dispatch('request-context', { [propertyName]: valueOrCallback }, true);
+	}
+
+	onRequestContext(event) {
```

Comment:
@quarkus This is still using and mixing provide/inject with context wording...

Reply:
Yes .. but thats actually intentional to use the requestContext Protocol "under the hood" and to interface it through the inject/provideProperties to the developer.

Reply:
Alright